### DISCUSSION TITLE
Newbie Question // Issues with building

### DISCUSSION THREAD

**Discussion by TehPig** [Minestom/Minestom#1853] [Help] [answered]
Hey there,

I am trying to setup a basic server to try out, but I am currently having issues with building the jar file.
The issue seems to originate from the `build.gradle.kts` file and I tried to follow the guide provided [here](https://wiki.minestom.net/setup/your-first-server)

If you need the file structure or any other additional information let me know.
Any help would be really appreciated.

**Comment by TehPig** [comment]
Apparently the `.kts` file type is what made the build unsuccessful.

**Comment by TehPig** [comment]
I actually have another question: In which directory can I find the built jar file?

**Reply by GoldenCookieGIT** [reply]
if you didn't change the output directory in the build script, it should be in ``build/libs`` in the root directory of the project

**Reply by TehPig** [reply]
there is no build/libs directory though

**Reply by TehPig** [reply]
I do not think that options from other java projects could have been moved to the minestom server.
As for the build.gradle file, here is the cod snipper
```js
plugins {
    id 'java'
    id "com.github.johnrengelman.shadow" version "7.1.1"
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
    maven { url 'https://jitpack.io' }
}

dependencies {
    // WARNING: outdated version, replace it to the latest
    // Reminder: can be found at https://jitpack.io/#Minestom/Minestom
    implementation 'com.github.Minestom:Minestom:-SNAPSHOT'
}

jar {
    manifest {
        // WARNING: change the 'Main-Class' value
        attributes 'Main-Class': 'fr.themode.minestom.MinestomTest',
                "Multi-Release": true
    }
}
```

This is my current file structure
![image](https://github.com/Minestom/Minestom/assets/34095153/eb4226bd-dcb2-4650-83a5-507224edb727)

**Reply by GoldenCookieGIT** [reply]
open up the orange build folder, there should be a libs or lib folder there

**Reply by GoldenCookieGIT** [reply]
also change the main class name in the main-class manifest to your main class

**Reply by TehPig** [reply]
There is no lib or libs directory inside the builds folder for me.
As for the main class, I have named my main file to Main.java, so the class name is Main

**Comment by TehPig** [comment]
Apparently I managed to make the build, but I had to use the command line to do it.
IntelliJ's build button seemed to do nothing.

**Reply by zax71** [reply, marked as answer]
You can make a custom build script using IntelliJ, click on the drop down next to the play button and select gradle then set build as the command
